BitwardenShared/Core/Platform/Services/AuthenticatorSyncService.swift

```diff
@@ -102,9 +106,33 @@ actor DefaultAuthenticatorSyncService: NSObject, AuthenticatorSyncService {
     public func start() async {
         guard !started else { return }
         started = true
-        if await configService.getFeatureFlag(FeatureFlag.enableAuthenticatorSync,
-                                              defaultValue: false) {
-            subscribeToAppState()
+
+        guard await configService.getFeatureFlag(FeatureFlag.enableAuthenticatorSync,
+                                                 defaultValue: false) else {
+            return
+        }
+
+        syncSettingSubscriberTask = Task {
+            for await (userId, _) in await self.stateService.syncToAuthenticatorPublisher().values {
+                guard let userId else { continue }
+
+                do {
+                    try await determineSyncForUserId(userId)
+                } catch {
+                    errorReporter.log(error: error)
+                }
+            }
+        }
+        vaultUnlockSubscriberTask = Task {
+            for await vaultStatus in await self.vaultTimeoutService.vaultLockStatusPublisher().values {
+                guard let vaultStatus else { continue }
+
+                do {
+                    try await determineSyncForUserId(vaultStatus.userId)
+                } catch {
+                    errorReporter.log(error: error)
+                }
+            }
         }
```

Comment:
🤔 Would it make sense to use just one `Task` with both publishers being combined using something like `Publishers.CombineLatest`?

Reply:
I was thinking of going that route, but that puts the two into a tuple. The problem there is we could have updates for multiple different accounts, which then would be tupled together and we wouldn't be able to know which of the two is the actionable side of that. It also seems weird to have the syncing for `account1` tupled together with the vault status of `account2`.

If you can think of a way to handle this, I'd definitely be open. I'm less well-versed on combine. But the end result of both of these paths is just calling `determineSyncForUserId(userId)` and handling errors. So I'd rather have them all handled in one task if we can find a way.

Reply:
Perhaps using `merge` if we first translated them both to just publishing a `String?` of their username?